core/server/src/main/java/alluxio/master/file/meta/InodeTree.java

```diff
@@ -946,9 +961,11 @@ private TraversalResult traverseToInode(String[] pathComponents, LockMode lockMo
         if (pathComponents[0].equals("")) {
           if (getLockModeForComponent(0, pathComponents.length, lockMode, lockHints)
               == LockMode.READ) {
-            lockList.lockRead(mRoot);
+            // Do not need to check parent for root.
+            lockList.lockReadAndCheckParent(mRoot, null);
```

Comment:
it might make sense to define SKIP_ROOT_PARENT_CHECK = null someplace and use it instead of the comment